.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,53 @@
+name: build_and_test
+
+on:
+  push:
+    branches:
+      - main
+      - master # TODO: Remove this in favor of main eventually?
+      - next
+  pull_request:
```

Comment:
Would be good if we can run lint, typecheck and test on pull requests - to know things are going to break before merging (and later we can also disallow merging if the build fails)

Reply:
It should run on pull requests once merged, or maybe you need to enable actions in your repository settings